tests/functional/test_freeze.py

```diff
@@ -258,41 +259,31 @@ def test_freeze_mercurial_clone(script, tmpdir):
     _check_output(result, expected)
 
 
-@pytest.mark.network
 def test_freeze_bazaar_clone(script, tmpdir):
     """
     Test freezing a Bazaar clone.
 
     """
-
-    checkout_path = local_checkout(
-        'bzr+http://bazaar.launchpad.net/%7Edjango-wikiapp/django-wikiapp/'
-        'release-0.1',
-        tmpdir.join("cache"),
-    )
-    # bzr internally stores windows drives as uppercase; we'll match that.
+    try:
+        checkout_path = _create_test_package(script, vcs='bazaar')
+    except OSError as e:
+        pytest.skip('Invoking `bzr` failed: %s' % e)
```

Comment:
Good point. I'll add:

```python
@pytest.mark.bzr
```

and then folks can skip the test by doing for example:

```
py.test tests/functional/test_freeze.py -k 'not bzr'
```

How does that sound?

On Wed, Mar 11, 2015 at 9:57 AM, Paul Moore <notifications@github.com>
wrote:

> In tests/functional/test_freeze.py
> <https://github.com/pypa/pip/pull/2515#discussion_r26231414>:
>
> >  def test_freeze_bazaar_clone(script, tmpdir):
> >      """
> >      Test freezing a Bazaar clone.
> >
> >      """
> > -
> > -    checkout_path = local_checkout(
> > -        'bzr+http://bazaar.launchpad.net/%7Edjango-wikiapp/django-wikiapp/'
> > -        'release-0.1',
> > -        tmpdir.join("cache"),
> > -    )
> > -    # bzr internally stores windows drives as uppercase; we'll match that.
> > +    try:
> > +        checkout_path = _create_test_package(script, vcs='bazaar')
> > +    except OSError as e:
> > +        pytest.skip('Invoking `bzr` failed: %s' % e)
>
> So how does one say "I don't have bzr installed and don't intend to
> install it, please skip these tests"? Presumably it's a matter of setting
> some sort of "resource not available" flag on the test invocation? Could a
> note be added in docs/development.rst?
>
> Or are we just saying that developers *must* have all 4 VCSs installed to
> run the tests? (If so, that's fine for now but I'll probably revisit it
> when we get Windows tests going ;-))
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pypa/pip/pull/2515/files#r26231414>.
>